Subject: [PATCH] Fix NPE with void android.view.VelocityTracker.clear()

 E AndroidRuntime: FATAL EXCEPTION: main
 E AndroidRuntime: Process: com.android.settings, PID: 30816
 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 
 'void android.view.VelocityTracker.clear()' on a null object reference

diff --git a/core/java/android/widget/ScrollView.java b/core/java/android/widget/ScrollView.java
index 97d32f1..7993393 100644
--- a/core/java/android/widget/ScrollView.java
+++ b/core/java/android/widget/ScrollView.java
@@ -692,7 +692,9 @@
                     if (overScrollBy(0, deltaY, 0, mScrollY, 0, range, 0, mOverscrollDistance, true)
                             && !hasNestedScrollingParent()) {
                         // Break our velocity if we hit a scroll barrier.
-                        mVelocityTracker.clear();
+                        if (mVelocityTracker != null) {
+                            mVelocityTracker.clear();
+                        }
                     }
 
                     final int scrolledDeltaY = mScrollY - oldY;
